src/models/beach.ts

```diff
@@ -27,13 +31,12 @@ const schema = new mongoose.Schema(
   {
     toJSON: {
       transform: (_, ret): void => {
-        ret.id = ret._id;
+        ret.id = ret._id.toString();
         delete ret._id;
         delete ret.__v;
       },
     },
   }
 );
 
-interface BeachModel extends Omit<Beach, '_id'>, Document {}
-export const Beach: Model<BeachModel> = mongoose.model('Beach', schema);
+export const Beach = mongoose.model<Beach>('Beach', schema);
```

Comment:
[No action]

Vou deixar esse comentário no model porque as vezes o modelo é também usado como uma entidade de domínio contendo lógica (ex: RoR). Porém, o comentário não tem nada haver com as mudanças feitas aqui. 

Como eu disse em outro comentário, eu não sei o quanto tu vai ir adentro do clean architecture. Caso tu te aprofunde mais, talvez seja interessante trazer exemplos de como usar entiades de domínio para garantir as regras de negócio. Eu to tentando olhar para a tua aplicação e pensar em entidades que façam sentido, logo diagrama abaixo faz sentido para ti?

![image](https://user-images.githubusercontent.com/3832154/170988948-9520e202-0335-4009-aef4-83fec673e326.png)

Eu achei o [RatingService](https://github.com/waldemarnt/node-typescript-api/blob/c1425a60ccb04af7ddf98e6fdb27e32425eba43b/src/services/rating.ts) da um excelente exemplo de entidade de domínio já que ele é 100% sobre regras de negócio e tem zero side-effects. 

Transformar o Beach, Forecast e Rating me parece que teria um impacto interessante na arquitetura e design da tua aplicação:
1. A lógica de negócio seria 100% testável sem depender do banco já que as entidades de domínio seriam puras.
2. O teu repositório poderia passar a receber uma entidade ao invés de um object. Isso garantiria uma maior consistencia no estado do banco com o estado da applicação já que:
Isso garantiria que:
   1. Os dados no banco estariam no formato certo graças aos schemas MongoDB;
   2. Os dados no banco estariam também validados com as regras de negócio.
 3. A tua camada de serviço seria o [imperative shell](https://www.kennethlange.com/functional-core-imperative-shell/) da tua aplicação

Reply:
Sim concordo, acho que ficaria legal, vou fazer isso como parte do próximo conteúdo e a gente arquiteta junto. Nesse eu tenho uma arquitetura simples com services e transito uma estrutura de dados simples entre eles, objeto, como tu falou. Mas agora adicionando repositórios acho que entidade vai cair bem.